include/sys/param.h

```diff
@@ -43,6 +43,12 @@
 #  define MAX(a,b)      (((a) > (b)) ? (a) : (b))
 #endif  /* MAX */
 
+/* Macros for number of items. */
+
+#ifndef nitems
```

Comment:
it was obvious  ARRAY_SIZE was macro when read in code. This look to be a function call.  It may not get found by someone adding code looking for and Array operation and then get duplicated again. 
Can you use ARRAY_SIZE?

Reply:
Yes, agree with you, ARRAY_SIZE() looks better than nitems(), let me keep the ARRAY_SIZE

Reply:
I suggest to keep both then. Having `nitems` declared might be useful when integrating 3rd party apps.

Reply:
> I suggest to keep both then. Since `nitems` is commonly defined in `sys/param.h`, it might be useful when integrating 3rd party apps.

@anchao - Were you going to keep both nitems and ARRAY_SIZE?

Reply:
ARRAY_SIZE conflict with the 3rd party library(e.g. libuv), @anchao remove it from the change.

Reply:
@davids5 do you have more concern?

Reply:
Just the same one. If we have a #if defined(ARRAY_SIZE) won't that solve the 3rd party library issues?

Reply:
no, since the sequence is that the 3rd party library include sys/param.h first and then define ARRAY_SIZE unconditionally:(. The check in our header file can't avoid the duplication definition error.

Reply:
Can we add a comment to `nitems ` with `ARRAY_SIZE` in the text?

Reply:
could you explain more?

Reply:
see  https://github.com/apache/nuttx/pull/8414/files#r1096665233